cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -0,0 +1,126 @@
+      "wdlDependencies" -> workflow.zippedImports
+    ) collect { case (name, Some(file)) => Multipart.FormData.BodyPart.fromPath(name, MediaTypes.`application/zip`, file.path) }
+
+    val multipartFormData = Multipart.FormData((sourceBodyParts ++ zipBodyParts).toSeq : _*)
+    val requestEntity = multipartFormData.toEntity()
+
+    // Make a set of submissions that represent the batch (so we can zip with the results later):
+    val submissionSet = workflow.inputsBatch.map(inputs => WorkflowSubmission(workflow.wdl, Option(inputs), workflow.options, workflow.zippedImports, workflow.refreshToken))
+
+    makeRequest[List[CromwellStatus]](HttpRequest(HttpMethods.POST, batchSubmitEndpoint, List.empty[HttpHeader], requestEntity)) map { statuses =>
+      val zipped = submissionSet.zip(statuses)
+      zipped map { case (submission, status) =>
+        SubmittedWorkflow(WorkflowId.fromString(status.id), cromwellUrl, submission)
+      }
+    }
+  }
+
+  def abort(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowStatus] = getRequest[CromwellStatus](abortEndpoint(workflowId)) map WorkflowStatus.apply
+  def status(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowStatus] = getRequest[CromwellStatus](statusEndpoint(workflowId)) map WorkflowStatus.apply
+  def metadata(workflowId: WorkflowId)(implicit ec: ExecutionContext): Future[WorkflowMetadata] = getRequest[String](metadataEndpoint(workflowId)) map WorkflowMetadata
+  def backends(implicit ec: ExecutionContext): Future[CromwellBackends] = getRequest[CromwellBackends](backendsEndpoint)
+
+  /**
+    *
+    * @tparam A The type of response expected. Must be supported by an implicit unmarshaller from ResponseEntity.
+    */
+  private def makeRequest[A](request: HttpRequest)(implicit um: Unmarshaller[ResponseEntity, A], ec: ExecutionContext): Future[A] = for {
+    response <- Http().singleRequest(request)
+    entity <- Future.fromTry(response.toEntity)
+    unmarshalled <- entity.to[A]
+  } yield unmarshalled
+
+  private def getRequest[A](uri: String)(implicit um: Unmarshaller[ResponseEntity, A], ec: ExecutionContext): Future[A] = makeRequest[A](HttpRequest(uri = uri))
+
+  private def insertSecrets(options: Option[String], refreshToken: Option[String]): Option[String] = {
+    import DefaultJsonProtocol._
+    val tokenKey = "refresh_token"
+
+    def addToken(optionsMap: Map[String, JsValue]): Map[String, JsValue] = {
+      refreshToken match {
+        case Some(token) if optionsMap.get(tokenKey).isDefined => optionsMap + (tokenKey -> JsString(token))
+        case _ => optionsMap
+      }
+    }
+
+    options map (o => addToken(o.parseJson.asJsObject.convertTo[Map[String, JsValue]]).toJson.toString)
+  }
+}
+
+object CromwellClient {
+  final implicit class EnhancedHttpResponse(val response: HttpResponse) extends AnyVal {
+    /**
+      * @throws RuntimeException if the response code is not 2xx, intended to be used inside mapping of Futures
```

Comment:
Also nothing else in the entire Cromwell ecosystem ever has `@throws` docs. I'll remove